data/src/main/java/org/cryptomator/data/cloud/crypto/CryptoImplVaultFormat7.kt

```diff
@@ -166,6 +165,18 @@ open class CryptoImplVaultFormat7 : CryptoImplDecorator {
 			}
 		}.map { node ->
 			ciphertextToCleartextNode(cryptoFolder, dirId, node)
+		}.onEach { cryptoNode ->
+			if (cryptoNode is CryptoFile && isImageMediaType(cryptoNode.name)) {
+				val cacheKey = generateCacheKey(cryptoNode.cloudFile)
+				cryptoNode.cloudFile.cloud?.type()?.let { cloudType ->
+					getLruCacheFor(cloudType)?.let { diskCache ->
+						val cacheFile = diskCache[cacheKey]
+						if (cacheFile != null) {
+							cryptoNode.thumbnail = cacheFile
+						}
+					}
+				}
+			}
```

Comment:
I'm a bit concerned about performance here, did you check how long it takes to open a folder containing 2000 images
1. with generated thumbnail
2. without generated thumbnail